src/spikeinterface/core/generate.py

```diff
@@ -801,11 +821,84 @@ def synthesize_random_firings(
     times = np.concatenate(times)
     labels = np.concatenate(labels)
 
-    sort_inds = np.argsort(times)
-    times = times[sort_inds]
-    labels = labels[sort_inds]
+    train_length = len(times)
+
+    spike_train = np.zeros(train_length, dtype=minimum_spike_dtype)
+    spike_train["sample_index"] = times
+    spike_train["unit_index"] = labels
+
+    spike_train = add_insertions(spike_train, insertions, insertions_replace, rng)
+
+    spike_train.sort(order=["sample_index"])
+
+    segment_length = train_length // segments
+    for i in range(train_length):
+        spike_train["segment_index"][i] = min(i // segment_length, segments - 1)
+
+    return spike_train
+
+
+def add_insertions(spike_train, insertions, insertions_replace, rng=None, seed=None):
+    """
+    Add specified insertions into a list of times and unit labels.
+
+    Parameters
+    ----------
+    spike_train: np.ndarray
+        Structured numpy array ("sample_index", "unit_index", "segment_index").
+    insertions: array-like
+        List of manually inserted spikes, each spike formatted as [sample index, unit index]. Intended for
+        use in testing.
+    insertions_replace: bool
+        If True, randomly replace generated spikes. If False, add to generated spikes.
+    rng: numpy.random.Generator
+        A random number generator
+    seed: int, default: None
+        seed for the generator
+
+    Returns
+    -------
+    spike_train: np.ndarray
+        Structured numpy array ("sample_index", "unit_index", "segment_index").
+        Including insertions.
+
+    """
+
+    if insertions is None:
+        return spike_train
+
+    new_spike_train = spike_train
+
+    if rng is None:
+        rng = np.random.default_rng(seed=seed)
+
+    insertions = get_structured_insertions(insertions)
+    if insertions_replace:
+        replacement_indices = rng.choice(len(spike_train), len(insertions), replace=False)
+        new_spike_train[replacement_indices] = insertions
+    else:
+        new_spike_train = np.append(spike_train, insertions)
+
+    return new_spike_train
+
+
+def get_structured_insertions(insertions):
+
+    structured_insertions = np.zeros(
+        len(insertions), dtype=[("sample_index", "int64"), ("unit_index", "int64"), ("segment_index", "int64")]
```

Comment:
I suggest using the `minimum_spike_dtype` so  this is properly coupled or defining it and adding a comment exlpaining why this is mean to be different.